modules/cnf-image-based-upgrade-generate-seed-image.adoc

```diff
@@ -0,0 +1,184 @@
+// Module included in the following assemblies:
+// * edge_computing/image-based-upgrade/cnf-preparing-for-image-based-upgrade.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="ztp-image-based-upgrade-seed-generation_{context}"]
+= Generating a seed image with the {lcao}
+
+Use the {lcao} to generate the seed image with the `SeedGenerator` CR. The Operator checks for required system configurations, performs any necessary system cleanup before generating the seed image, and launches the image generation. The seed image generation includes the following tasks:
+
+* Stopping cluster operators
+* Preparing the seed image configuration
+* Generating and pushing the seed image to the image repository specified in the `SeedGenerator` CR
+* Restoring cluster operators
+* Expiring seed cluster certificates
+* Generating new certificates for the seed cluster
+* Restoring and updating the `SeedGenerator` CR on the seed cluster
+
+.Prerequisites
+
+* Configure a shared container directory on the seed cluster.
+* Install the OADP Operator and the {lcao} on the seed cluster.
+
+.Procedure
+
+. Detach the cluster from the hub to delete any cluster-specific resources from the seed cluster that must not be in the seed image:
+
+.. If you are using {rh-rhacm}, manually detach the seed cluster by running the following command:
++
+[source,terminal]
+----
+$ oc delete managedcluster sno-worker-example
+----
+
+... Wait until the `ManagedCluster` CR is removed. Once the CR is removed, create the proper `SeedGenerator` CR. The {lcao} cleans up the {rh-rhacm} artifacts.
+
+.. If you are using GitOps ZTP, detach your cluster by removing the seed cluster's `SiteConfig` CR from the `kustomization.yaml`:
+
+... Remove your seed cluster's `SiteConfig` CR from the `kustomization.yaml`.
++
+[source,yaml]
+----
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+generators:
+#- example-seed-sno1.yaml
+- example-target-sno2.yaml
+- example-target-sno3.yaml
+----
+
+... Commit the `kustomization.yaml` changes in your Git repository and push the changes.
+
++
+The ArgoCD pipeline detects the changes and removes the managed cluster.
+
+. Create the `Secret`.
+
+.. Create the authentication file by running the following command:
++
+--
+.Authentication file
+[source,terminal]
+----
```

Comment:
This appears to copy correct. Would you happen to know offhand if copying and pasting this command in your terminal will run all three steps correct? I'm assuming that it will. If it does not, you might consider having three separate code blocks for this content.

Reply:
I just ran this in my terminal, of course podman couldn't log into the nonexistent registry but all of them executed, so it works :)